### PR TITLE
Improve ARM64 pseudo disassembly with compound assignments ##arch 

### LINKED ISSUE #15947: pseudo for arm64 can be improved (+= vs +)
```
> wx 297d0a9b
> pd 1
mul x9, x9, x10
> e asm.pseudo = 1
> pd 1
x9 = x9 + x10
```

expected:

```
x9 += x10
```

### PR DESCRIPTION
**Pull request by vicky-dx**
## Description
Improves ARM64 pseudo disassembly by converting verbose compound assignment patterns to compact C-style operators.

## Changes
- **Before**: `x9 = x9 * x10`
- **After**: `x9 *= x10`

Supports 7 operators: `*=`, `/=`, `&=`, `|=`, `^=`, `+=`, `-=`

## Implementation
- Post-processing pattern matching in `libr/arch/p/arm/pseudo.c`
- Detects "reg = reg OP val" patterns and converts to "reg OP= val"
- Handles edge cases: w-registers, unsigned division, type casts

## Testing
- Added comprehensive test suite: `test/db/cmd/cmd_pseudo_arm64`
- 18 test cases covering all operators and edge cases
- All tests passing

## Fixes
Closes #15947

### CODE REVIEW

File: libr/arch/p/arm/pseudo.c [hunk 1]
```diff
@@ -289,6 +300,108 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
 		s = r_str_replace (s, " lsr ", " >> ", 1);
 		s = r_str_replace (s, "+ -", "- ", 1);
 		s = r_str_replace (s, "- -", "+ ", 1);
+
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]: empty line with training whitespaces

File: libr/arch/p/arm/pseudo.c [hunk 2]
```diff
@@ -289,6 +300,108 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
 		s = r_str_replace (s, "- -", "+ ", 1);
+		
+		// Simplify "reg = reg OP val" to "reg OP= val"
+		// Pattern: "x9 = x9 * x10" -> "x9 *= x10"
+		char *eq = strstr (s, " = ");
+		if (eq) {
+			char *start = s;
+			char *op_pos = NULL;
+			// Find operator after '='
+			if ((op_pos = strstr (eq + 3, " * "))) {
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]: all thsese cases look like they can be implemented in a single function instead of repeating the same code logic for each case. make a separate function and reuse logic

File: libr/arch/p/arm/pseudo.c [hunk 3]
```diff
@@ -289,6 +300,42 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+		// Simplify "reg = reg OP val" to "reg OP= val"
+		// Pattern: "x9 = x9 * x10" -> "x9 *= x10"
+		char *eq = strstr (s, " = ");
+		if (eq) {
+			char *start = s;
+			// List of operators to check: *, /, &, |, ^, +, -
+			const char *operators[] = { " * ", " / ", " & ", " | ", " ^ ", " + ", " - ", NULL };
+			const char *compound[] = { " *= ", " /= ", " &= ", " |= ", " ^= ", " += ", " -= " };
+
+			for (int i = 0; operators[i]; i++) {
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
			int;
			for (i = 0; operators[i]; i++) {
```

File: libr/arch/p/arm/pseudo.c [hunk 4]
```diff
@@ -289,6 +300,42 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+				char *op_pos = strstr (eq + 3, operators[i]);
+				if (op_pos) {
+					char reg[32] = {0};
+					size_t reg_len = eq - start;
+					if (reg_len < sizeof (reg)) {
+						strncpy (reg, start, reg_len);
+						// Check for "(unsigned) reg /" pattern (special case for division)
+						char *check_pos = eq + 3;
+						if (i == 1 && !strncmp (check_pos, "(unsigned) ", 11)) {
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
						if (i == 1 && r_str_startswith (check_pos, "(unsigned) ")) {
```

File: libr/arch/p/arm/pseudo.c [hunk 5]
```diff
@@ -289,6 +300,42 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+			const char *operators[] = { " * ", " / ", " & ", " | ", " ^ ", " + ", " - ", NULL };
+			const char *compound[] = { " *= ", " /= ", " &= ", " |= ", " ^= ", " += ", " -= " };
+
+			for (int i = 0; operators[i]; i++) {
+				char *op_pos = strstr (eq + 3, operators[i]);
+				if (op_pos) {
+					char reg[32] = {0};
+					size_t reg_len = eq - start;
+					if (reg_len < sizeof (reg)) {
+						strncpy (reg, start, reg_len);
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
						r_str_ncpy (reg, start, reg_len);
```

File: libr/arch/p/arm/pseudo.c [hunk 6]
```diff
@@ -289,6 +300,42 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+		if (eq) {
+			char *start = s;
+			// List of operators to check: *, /, &, |, ^, +, -
+			const char *operators[] = { " * ", " / ", " & ", " | ", " ^ ", " + ", " - ", NULL };
+			const char *compound[] = { " *= ", " /= ", " &= ", " |= ", " ^= ", " += ", " -= " };
+
+			for (int i = 0; operators[i]; i++) {
+				char *op_pos = strstr (eq + 3, operators[i]);
+				if (op_pos) {
+					char reg[32] = {0};
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
					char reg[32];
```

File: libr/arch/p/arm/pseudo.c [hunk 7]
```diff
@@ -288,6 +300,43 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+		// Simplify "reg = reg OP val" to "reg OP= val"
+		// Pattern: "x9 = x9 * x10" -> "x9 *= x10"
+		char *eq = strstr (s, " = ");
+		if (eq) {
+			char *start = s;
+			// List of operators to check: *, /, &, |, ^, +, -
+			const char *operators[] = { " * ", " / ", " & ", " | ", " ^ ", " + ", " - ", NULL };
+			const char *compound[] = { " *= ", " /= ", " &= ", " |= ", " ^= ", " += ", " -= " };
+
+			int;
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
			int i;
```

File: libr/arch/p/arm/pseudo.c [hunk 8]
```diff
@@ -288,6 +300,43 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+				char *op_pos = strstr (eq + 3, operators[i]);
+				if (op_pos) {
+					char reg[32];
+					size_t reg_len = eq - start;
+					if (reg_len < sizeof (reg)) {
+						r_str_ncpy (reg, start, reg_len);
+						// Check for "(unsigned) reg /" pattern (special case for division)
+						char *check_pos = eq + 3;
+						if (i == 1 && r_str_startswith (check_pos, "(unsigned) ")) {
+							check_pos += 11; // skip "(unsigned) "
```
Comment (trufae) [inline libr/arch/p/arm/pseudo.c]:
```suggestion
							check_pos += strlen ("(unsigned) ")
```

**Comment by trufae** [comment]
it would be great if we can generalize this stuff for the r_str_pseudo api, the plan is to avoid having custom asm parsers in every arch and just have some rules that deifne the way instructions are constructed in there. and compound assigments is something tha may affect all archs too. so having this code adapted to the r_str_pseudo will help transition more archs to use this api. thanks for providing all the tests!

File: libr/arch/p/arm/pseudo.c [hunk 9]
```diff
@@ -289,6 +300,43 @@ static char *parse(RAsmPluginSession *aps, const char *data) {
...
+				char *op_pos = strstr (eq + 3, operators[i]);
+				if (op_pos) {
+					char reg[32];
+					size_t reg_len = eq - start;
+					if (reg_len < sizeof (reg)) {
+						r_str_ncpy (reg, start, reg_len);
+						// Check for "(unsigned) reg /" pattern (special case for division)
+						char *check_pos = eq + 3;
+						if (i == 1 && r_str_startswith (check_pos, "(unsigned) ")) {
+							check_pos += strlen ("(unsigned) ")
```
Comment (AbhiTheModder) [inline libr/arch/p/arm/pseudo.c:324]: Missing a `;`
